### PR TITLE
Logs for partial flushing

### PR DESCRIPTION
**Pull request by wantsui**
## Description

The purpose of this PR to add additional logging when [partial flushing has been configured](https://ddtrace.readthedocs.io/en/latest/advanced_usage.html?highlight=configure#ddtrace.Tracer.configure).

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: ddtrace/internal/debug.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def collect(tracer):
             integration_configs[module] = "N/A"
 
     pip_version = packages_available.get("pip", "N/A")
-
+
```
Comment (brettlangdon) [inline ddtrace/internal/debug.py:102]: this change seems unrelated, please be sure to run `black` on any files you change.

`riot run -s fmt`

File: ddtrace/internal/debug.py [hunk 2]
```diff
@@ -134,4 +134,6 @@ def collect(tracer):
         global_tags=os.getenv("DD_TAGS", ""),
         tracer_tags=tags_to_str(tracer.tags),
         integrations=integration_configs,
+        partial_flush_enabled=tracer._partial_flush_enabled,
+        partial_flush_min_spans=tracer._partial_flush_min_spans,
```
Comment (brettlangdon) [inline ddtrace/internal/debug.py:138]: I don't love reading this private property... but it does seem the easiest way to do it

File: ddtrace/internal/processor/trace.py [hunk 3]
```diff
@@ -141,6 +141,9 @@ def on_span_finish(self, span):
                 else:
                     finished = trace_spans
 
+                if self._partial_flush_enabled and trace.num_finished >= self._partial_flush_min_spans:
+                    log.debug("Partially flushing %d spans for trace %d", len(trace.spans), span.trace_id)
```
Comment (brettlangdon) [inline ddtrace/internal/processor/trace.py:145]:
```suggestion
                    log.debug("Partially flushing %d spans for trace %d", len(finished), span.trace_id)
```

File: ddtrace/internal/processor/trace.py [hunk 4]
```diff
@@ -141,6 +141,9 @@ def on_span_finish(self, span):
...
+
                 trace.num_finished -= len(finished)
```
Comment (brettlangdon) [inline ddtrace/internal/processor/trace.py:147]: since we are doing `len(finished)` twice, might be worth doing it in a variable?

```python
num_finished = len(finished)

if partial_flushing:
    log.debug(num_finished)

trace.num_finished -= num_finished
```

File: ddtrace/internal/processor/trace.py (hunk 3 above)
Comment (brettlangdon) [reply ddtrace/internal/processor/trace.py:145]: we should also add a test case for this log line as well

File: tests/integration/test_debug.py [hunk 5]
```diff
@@ -371,3 +371,20 @@ def test_debug_span_log():
     p.wait()
     stderr = p.stderr.read()
     assert b"finishing span name='span'" in stderr
+
+
+def test_partial_flush_log():
```
Comment (brettlangdon) [inline tests/integration/test_debug.py:376]: can we also add a test for when this is configured via environment variables?

File: ddtrace/internal/processor/trace.py [hunk 6]
```diff
@@ -141,6 +141,9 @@ def on_span_finish(self, span):
                 else:
                     finished = trace_spans
 
+                if self._partial_flush_enabled and trace.num_finished >= self._partial_flush_min_spans:
```
Comment (brettlangdon) [inline ddtrace/internal/processor/trace.py:144]: we should probably do this check only once

```python
should_partial_flush = self._partial_flush_enabled and trace.num_finished >= self._partial_flush_min_spans
if trace.num_finished == len(trace.spans) or should_partial_flush:
    # do stuff

    if should_partial_flush:
        log.debug(num_finished)
```

Comment (brettlangdon) [reply ddtrace/internal/processor/trace.py:144]: what you haven't isn't too bad performance wise, just thinking about legibility and in the future if we need to adjust the logic for "should partial flush" we can do so in only one place.

File: ddtrace/internal/debug.py (hunk 1 above)
Comment (wantsui) [reply ddtrace/internal/debug.py]: Thanks! I ran this is the latest set of commit and I think this got removed :)

File: ddtrace/internal/processor/trace.py (hunk 3 above)
Comment (wantsui) [reply ddtrace/internal/processor/trace.py]: I started writing a test for `Partially flushing %d spans for trace` in the test_integration.py file, but the mock call seems to return an empty log. I'm not sure why though.

File: ddtrace/internal/debug.py (hunk 2 above)
Comment (Kyle-Verhoog) [reply ddtrace/internal/debug.py:138]: Don't think we have an alternative here since it is an internal + undocumented API. This one isn't too bad all things considered

**Review by Kyle-Verhoog — DISMISSED** [review]
lgtm!

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by P403n1x87 — APPROVED** [review]
